beacon-chain/p2p/peers/status.go

```diff
@@ -0,0 +1,311 @@
+// Package peers provides information about peers at the Ethereum protocol level.
+// "Protocol level" is the level above the network level, so this layer never sees or interacts with (for example) hosts that are
+// uncontactable due to being down, firewalled, etc.  Instead, this works with peers that are contactable but may or may not be of
+// the correct fork version, not currently required due to the number of current connections, etc.
+//
+// A peer can have one of a number of states:
+//
+// - disconnected if we are not able to talk to the remote peer
+// - connecting if we are attempting to be able to talk to the remote peer
+// - connected if we are able to talk to the remote peer
+// - disconnecting if we are attempting to stop being able to talk to the remote peer
+//
+// For convenience, there are two aggregate states expressed in functions:
+//
+// - active if we are connecting or connected
+// - inactive if we are disconnecting or disconnected
+//
+// Peer information is persistent for the run of the service.  This allows for collection of useful long-term statistics such as
+// number of bad responses obtained from the peer, giving the basis for decisions to not talk to known-bad peers.
+package peers
```

Comment:
Can you add tests for this please? There is some filtering and other logic that should have unit tests.

Reply:
The `peers` code now has 100% coverage.